spec/datadog/profiling/crashtracker_spec.rb

```diff
@@ -0,0 +1,201 @@
+require 'datadog/profiling/spec_helper'
+require 'datadog/profiling/crashtracker'
+
+require 'webrick'
+
+RSpec.describe Datadog::Profiling::Crashtracker do
+  before { skip_if_profiling_not_supported(self) }
+
+  let(:exporter_configuration) { [:agent, 'http://localhost:6006'] }
+
+  let(:crashtracker_options) do
+    {
+      exporter_configuration: exporter_configuration,
+      tags: { 'tag1' => 'value1', 'tag2' => 'value2' },
+      upload_timeout_seconds: 123,
+    }
+  end
+
+  subject(:crashtracker) { described_class.new(**crashtracker_options) }
+
+  describe '#start' do
+    subject(:start) { crashtracker.start }
+
+    context 'when _native_start_or_update_on_fork raises an exception' do
+      it 'logs the exception' do
+        expect(described_class).to receive(:_native_start_or_update_on_fork) { raise 'Test failure' }
+        expect(Datadog.logger).to receive(:error).with(/Failed to start crash tracking: Test failure/)
+
+        start
+      end
+    end
+
+    context 'when path_to_crashtracking_receiver_binary is nil' do
+      subject(:crashtracker) { described_class.new(**crashtracker_options, path_to_crashtracking_receiver_binary: nil) }
+
+      it 'logs a warning' do
+        expect(Datadog.logger).to receive(:warn).with(/no path_to_crashtracking_receiver_binary was found/)
+
+        start
+      end
+    end
+
+    context 'when ld_library_path is nil' do
+      subject(:crashtracker) { described_class.new(**crashtracker_options, ld_library_path: nil) }
+
+      it 'logs a warning' do
+        expect(Datadog.logger).to receive(:warn).with(/no ld_library_path was found/)
+
+        start
+      end
+    end
+
+    it 'starts the crash tracker' do
+      start
+
+      expect(`pgrep -f libdatadog-crashtracking-receiver`).to_not be_empty
+
+      crashtracker.stop
+    end
+
+    context 'when calling start multiple times in a row' do
+      it 'only starts the crash tracker once' do
+        3.times { crashtracker.start }
+
+        expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 1
+
+        crashtracker.stop
+      end
+    end
+
+    context 'when upload_timeout_seconds is not an Integer' do
+      let(:crashtracker_options) { { **super(), upload_timeout_seconds: 12.34 } }
+
+      it 'converts it to an Integer before calling _native_start_or_update_on_fork' do
+        expect(described_class)
+          .to receive(:_native_start_or_update_on_fork).with(hash_including(upload_timeout_seconds: 12))
+
+        start
+      end
+    end
+  end
+
+  describe '#reset_after_fork' do
+    subject(:reset_after_fork) { crashtracker.reset_after_fork }
+
+    context 'when called in a fork' do
+      before { crashtracker.start }
+      after { crashtracker.stop }
+
+      it 'starts a second crash tracker for the fork' do
+        expect_in_fork do
+          crashtracker.reset_after_fork
+
+          expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 2
```

Comment:
Is it possible to have other processes running that could be captured here?

Reply:
It is, although we never run our tests in parallel inside the same container in CI, so only if you're running the test suite locally would this potentially happen.

I've added an extra check at the beginning of these tests to check that no existing crash tracker was already running: 5431d2b07ce1b60039a4bf8680bd7e56e225656e .